Typo3 12 compatibility

hojalatheef opened the pull request.

hojalatheef committed:
[TASK] Replacing TYPO3 Core Deprecations and Compatibility Fixes

- Replaced Deprecations
- Started compatibility fixing for TYPO3 12 LTS

hojalatheef committed:
Replaced deprecated ElementBrowserController namespace with new one

hojalatheef committed:
DI for moduleTemplateController added and removed old BackendTemplate wrapper

hojalatheef committed:
Replaced execute() with executeQuery() and return type

hojalatheef committed:
Replaced execute and fetch methods

Classes/Hooks/CachingFrameworkLoggerHook.php
Classes/Hooks/ModifyElementInformationHook.php
Classes/Hooks/MoveTranslatedContentElementsHook.php
Classes/Routing/Aspect/PersistedTableMapper.php
Classes/Service/SolrService.php

hojalatheef committed:
Replaced FolderLinkHandler with new API function

\TYPO3\CMS\Recordlist\LinkHandler\FolderLinkHandler
replced with
\TYPO3\CMS\Backend\LinkHandler\FolderLinkHandler

hojalatheef committed:
Replaced FolderLinkHandler with new API function

\TYPO3\CMS\Recordlist\LinkHandler\FolderLinkHandler
replced with
 \TYPO3\CMS\Backend\LinkHandler\FolderLinkHandler

hojalatheef committed:
Deorecated caalls replaced

- replaced GeneralUtility::getUrl with @file_get_content
- replaced TYPO3_BE to ApplicationType method isFrontend()

hojalatheef committed:
[TASK] Test Suite Updated

- Updated Test Suite
- Added Testing Docker Configuration

hojalatheef committed:
Version details added

hojalatheef committed:
ReadMe updated with better badges and new CI status

hojalatheef committed:
[BUGFIX] Replaced Interface with Abstract class and replaced its implementation with extend keyword

hojalatheef committed:
Icon registration moved to Configuration/Icons.php

hojalatheef committed:
BackendModule registration moved to Configuration/Backend/Modules.php

hojalatheef committed:
BackendModule registration moved to Configuration/Backend/Modules.php

hojalatheef committed:
[TASK] Fixing Backend module configurations and paths

hojalatheef committed:
[TASK] Moved property DI to injectTraits

hojalatheef committed:
[TASK] Replaced deprecated AbstractMessage::MESSAGETYPE with ContextualFeedbackSeverity::Message

hojalatheef committed:
[TASK] Replaced GeneralUtility::_GP with ServerRequestInterface methods

hojalatheef committed:
[TASK] Fixed deprecations for BE Module templates and controller

hojalatheef committed:
Code optimization for RequestArgumentsTrait

hojalatheef committed:
[BUGFIX] Compatibility fixes commands done

hojalatheef committed:
Removed addHelpButton from header

hojalatheef committed:
[TASK] Replaced annotation property type return to strictly typed return types

hojalatheef committed:
[BUGFIX] Fixed property types as return and passed argument as casted (int)

hojalatheef committed:
[TASK] Replaced arguments with Traits

hojalatheef committed:
[TASK] Compatibility fixes

hojalatheef committed:
[TASK] Fixed issues with BE Modules for Solr module

hojalatheef committed:
[BUGFIX] Change redirect action with return

hojalatheef committed:
[TASK] Documentation updates and Code optimizations

- Release documentations updated
- Code optimizations done

hojalatheef committed:
[TASK] Migrating Test Cases to TYPO3 Testing Framework

- Unit Tests migrated to TYPO3 Testing Framework
- Functional Tests migrated to TYPO3 Testing Framework
- Removed obselete test cases which are removed because of removal

hojalatheef committed:
Added .env and testing related paths to ignore

hojalatheef committed:
[TASK] Migrated filelist context menu action for update/create metadata to version 12 LTS

- Removed jQuery dependency
- Replaced deprecations
- Removed old registration methods to updated

hojalatheef committed:
Updated Build path

hojalatheef committed:
Removed old ContextMenuActions.js

hojalatheef committed:
[BUGFIX] Fixed issue with argument missing in CLI run of Command Controller

hojalatheef committed:
[TASK] Removed obselete functionalities no more needed for 12 and cleanup

hojalatheef committed:
[TASK] Refactored ModifyElementInformationHook

hojalatheef committed:
Removed unwanted use of Connection class

hojalatheef committed:
Post Processing issue occures if the colPos 0 is handled

hojalatheef committed:
Removed deprecated files

hojalatheef committed:
Moved functionalities related to Link and Folder LinkHandler from Hooks to EventListener

hojalatheef committed:
Implementation for LinkHandlers done in EventListener

hojalatheef committed:
EventListener registration done in Services.yaml

hojalatheef committed:
Comment todo: for Hooks which needs to be removed after the release

hojalatheef committed:
Property type fix

hojalatheef committed:
[TASK] Refactoring and Code optimizations

hojalatheef committed:
Migrated SolrIndex.js from requireJS to ES6 Compatible by removing jquery support

hojalatheef committed:
Fixed CacheClearFullIndexForm to ES6 Removing jquery

hojalatheef committed:
[TASK] Refactor `typo3RequiredColumnsForFiles` functionality

- Deleted Classes/Backend/Browser/FileBrowser.php as now it is not compatible
- Moved logic to PSR-14 event `IsFileSelectableEvent`
- Removed unwanted configurations and settings from ext_localconf
- Updated documentations

hojalatheef committed:
Documentation added with update details and fixed issue with uid type

hojalatheef committed:
Fixed FTC issue with DI

hojalatheef committed:
Removed Link Handlers as the logic moved now to the PSR-14 event listener

hojalatheef committed:
Fixed formatting in documentation

sfroemkenjw commented on Classes/Backend/AbstractDocHeader.php:
```diff
@@ -23,32 +24,19 @@
-    protected $request;
-
-    public function __construct(Request $request, ViewInterface $view)
+    public function __construct(Request $request, ModuleTemplate $view)
```
Please move everything of that file into the SolrDocHeader class. Some classes were injected in the SolrDocHeader, other classes were injected in AbstractDocHeader, the constructor was in Abstract class, but because of given constructor arguments the DI does not work, which results in defining related classes within specific methods.

hojalatheef committed:
Small cleanup and optimizations

sfroemkenjw commented on Classes/Backend/SolrDocHeader.php:
```diff
@@ -10,20 +10,35 @@
     public function renderDocHeader(): void
     {
         // initialize UriBuilder
+        if (!($this->uriBuilder instanceof UriBuilder)) {
```
As it is part of the constructor arguments, these lines can be removed

sfroemkenjw commented on Classes/Backend/SolrDocHeader.php:
```diff
@@ -10,20 +10,35 @@
         $this->uriBuilder->setRequest($this->request);
 
         // Render Buttons
-        $this->addHelpButton();
```
why?

hojalatheef replied:
The implementation is deprecated and removed in TYPO3 12.

sfroemkenjw replied:
Not really. `addHelpButton` was a method which I have implemented on my own. It was part of the earlier AbstractDocHeader file which is now removed. All needed methods should be transferred to SolrDocHeader. Seems you missed that one: https://github.com/jweiland-net/jwtools2/pull/74/files/c0b89b40a22cfb70d333651ce29041f7f6968983#diff-1dec211ac487453d6d368447a3b94e6876dae1164fb2d3809ac6b8605713c6e9L64-L76

sfroemkenjw commented on Classes/Command/ConvertPlainPasswordToHashCommand.php:
```diff
@@ -156,13 +147,13 @@ protected function getDefaultHashInstance(string $mode): PasswordHashInterface
     }
 
-    protected function getStatementForUsers(string $tableName): Statement
+    protected function getStatementForUsers(string $tableName): Result
```
Rename method to `getQueryResultForUsers`

sfroemkenjw commented on Classes/Command/StatusReportCommand.php:
```diff
@@ -31,21 +30,20 @@ class StatusReportCommand extends Command
+
+    public function __construct()
+    {
+        parent::__construct();
```
I'm sure the parent constructor has a name argument

sfroemkenjw commented on Classes/Command/StatusReportCommand.php:
```diff
@@ -31,21 +30,20 @@ class StatusReportCommand extends Command
+    {
+        parent::__construct();
+
+        $this->taskRepository = GeneralUtility::makeInstance(SchedulerTaskRepository::class);
```
I prefer using a setter method which will be called by Services.yaml

sfroemkenjw commented on Classes/Command/StatusReportCommand.php:
```diff
@@ -192,11 +170,8 @@ protected function getContentOfRobotsTxt(Site $site): string
 
         if ($content === '' && !$this->input->hasOption('exclude-robots-txt-url-check')) {
             $base = rtrim((string)$site->getBase(), '/') . '/';
```
getBase can also return just `/`. Please try to find an API call to get full URL

hojaonline replied:
Confirmed by Slack to keep it as it is

hojalatheef replied:
@sfroemkenjw  Confirmed by Slack to keep it as it is

sfroemkenjw commented on Classes/Command/StatusReportCommand.php:
```diff
@@ -283,4 +258,43 @@ protected function getAllSites(): array
+    {
+        foreach ($tasks as $groups => $taskGroups) {
+            foreach ($taskGroups as $taskGroup) {
+                if (is_array($taskGroup['tasks']) && count($taskGroup['tasks'])) {
```
No need to count records. If empty, foreach will skip that automatically

sfroemkenjw commented on Classes/Command/StatusReportCommand.php:
```diff
@@ -283,4 +258,43 @@ protected function getAllSites(): array
         return GeneralUtility::makeInstance(SiteFinder::class)->getAllSites();
     }
+
+    protected function processTasks(array $tasks, array &$recurringTasks, int $yesterday, int $lastExecution): int
```
$yesterday is a fixed value. Move value into this method and remove from method args.

sfroemkenjw commented on Classes/Command/StatusReportCommand.php:
```diff
@@ -283,4 +258,43 @@ protected function getAllSites(): array
+        return $lastExecution;
+    }
+
+    protected function getTaskStatus(array $task, int $yesterday): string
```
$yesterday is a fixed value. Move value into this method and remove from method args.

sfroemkenjw commented on Classes/ContextMenu/ItemProviders/UpdateFileMetaDataProvider.php:
```diff
@@ -14,7 +14,8 @@
 
 /**
- * A command to execute extension updates realized with class.ext_update.php
+ * Adds a new entry ``Create/Update file metadata`` into context menu of filelist module to create a missing file
```
Please use normal quotes. These double quotes are only for MD or RST, but not useful for PHP comments

sfroemkenjw commented on Classes/Controller/AbstractController.php:
```diff
@@ -10,34 +10,41 @@
-     * @param ViewInterface $view The view to be initialized
      */
-    protected function initializeView(ViewInterface $view): void
+    protected function initializeView( TemplateView $view): void
```
Remove space

sfroemkenjw commented on Classes/Controller/Ajax/AjaxSolrController.php:
```diff
@@ -13,25 +13,27 @@
-        $postData = $request->getParsedBody();
-        $moduleData = $postData['tx_jwtools2'];
-
+        $moduleData = $request->getParsedBody();
```
Is `['tx_jwtools2']` missing here

hojalatheef replied:
Yeah I changed in the javascript

sfroemkenjw commented on Classes/Controller/Ajax/AjaxSolrController.php:
```diff
@@ -127,8 +137,7 @@ public function getProgressAction(ServerRequest $request): ResponseInterface
     {
-        $postData = $request->getParsedBody();
-        $moduleData = $postData['tx_jwtools2'];
+        $moduleData = $request->getParsedBody();
```
Is `['tx_jwtools2']` missing here

hojalatheef replied:
Yeah I changed in the javascript

sfroemkenjw commented on Classes/Controller/SolrController.php:
```diff
@@ -16,112 +16,80 @@
-/**
- * Class SolrController
- */
+#[AsController]
```
Is that really needed? As every class extending ActionController will be handled as Controller automatically

sfroemkenjw commented on Classes/Controller/ToolsController.php:
```diff
@@ -10,12 +10,19 @@
+use Psr\Http\Message\ResponseInterface;
+use TYPO3\CMS\Backend\Attribute\AsController;
+
+#[AsController]
```
IMO not needed as it extends ActionController

sfroemkenjw commented on Classes/Database/Query/QueryGenerator.php:
```diff
@@ -0,0 +1,74 @@
+    /**
+     * @throws Exception
+     */
+    public function getTreeList($id, $depth, $begin = 0, $permClause = ''): float|int|string
```
`float`?!?!

hojalatheef replied:
yeah, one of the function `abs($id)` have a return type float. But in our case, it is will be still int required.

sfroemkenjw commented on Classes/Hooks/ModifyElementInformationHook.php:
```diff
@@ -556,12 +506,8 @@ protected function getReferences(ServerRequestInterface $request): array
-     * @return string label
      */
-    protected function getLabelForTableColumn($tableName, $fieldName): string
+    protected function getLabelForTableColumnByTable(string $tableName, string $fieldName): string
```
In that case I prefer `getLabelForColumnByTable`

sfroemkenjw commented on Classes/Hooks/ModifyElementInformationHook.php:
```diff
@@ -721,13 +657,8 @@ protected function makeRef($table, $ref, ServerRequestInterface $request): array
-     * @return array
      */
-    protected function makeRefFrom($table, $ref, ServerRequestInterface $request): array
+    protected function makeRefFrom(string $table, int $ref, ServerRequestInterface $request): array
```
`int $ref` is correct? In old cold I have written `filename or uid`. Is string possible?

hojalatheef replied:
yeah I know what you mean the identifier format 1:filadmin/identifier. I changed it in the usage but missed in the argument type. Changed now.

sfroemkenjw commented on Classes/Task/ExecuteQueryTask.php:
```diff
@@ -37,14 +34,17 @@ public function execute(): bool
             foreach ($sqlQueries as $sqlQuery) {
-                $connection->query($sqlQuery)->execute();
+                // check $sqlQuery ends with a semi-colon otherwise add it
+                if (substr($sqlQuery, -1) !== ';') {
```
str_ends_with()

sfroemkenjw commented on Classes/Task/IndexQueueWorkerTaskAdditionalFieldProvider.php:
```diff
@@ -94,7 +94,7 @@ public function validateAdditionalFields(array &$submittedData, SchedulerModuleC
         /** @var IndexQueueWorkerTask $task */
-        $task->setDocumentsToIndexLimit($submittedData['documentsToIndexLimit']);
-        $task->setMaxSitesPerRun($submittedData['maxSitesPerRun']);
+        $task->setDocumentsToIndexLimit((int)$submittedData['documentsToIndexLimit']);
```
See line 81/82. The values already should be INT.

sfroemkenjw commented on Classes/Traits/InjectIconFactoryTrait.php:
```diff
@@ -0,0 +1,26 @@
+ */
+trait InjectIconFactoryTrait
+{
+    protected ?IconFactory $iconFactory = null;
```
nullable? Why. It's part of the initialization process?

hojalatheef replied:
![image](https://github.com/jweiland-net/jwtools2/assets/144037456/7f88a710-95b5-4e09-a543-91c554bb3a6c)

https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/DependencyInjection/Index.html#method-injection

sfroemkenjw commented on Classes/Traits/InjectUriBuilderTrait.php:
```diff
@@ -0,0 +1,26 @@
+ */
+trait InjectUriBuilderTrait
+{
+    protected ?UriBuilder $uriBuilder = null;
```
Why nullable. It's part of the initialization process.

hojalatheef replied:
![image](https://github.com/jweiland-net/jwtools2/assets/144037456/9c110140-d221-4468-a85d-2388db633757)

https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/DependencyInjection/Index.html#method-injection

@sfroemkenjw I had your same thought. As it is mentioned in the official documentation I keep it like this

sfroemkenjw commented on Classes/ViewHelpers/Format/HtmlViewHelper.php:
Remove that VH completely. The features are already part of TYPO3s Html Vh:
https://review.typo3.org/c/Packages/TYPO3.CMS/+/72803

sfroemkenjw commented on Configuration/Backend/Modules.php:
```diff
@@ -0,0 +1,30 @@
+                'overview',
+            ],
+            SolrController::class => [
+                'list','show', 'showIndexQueue', 'indexOneRecord', 'showClearIndexForm', 'clearIndex', 'showClearFullIndexForm',
```
Missing space before "show"

sfroemkenjw commented on Documentation/AdministratorManual/Upgrade/Index.rst:
```diff
@@ -0,0 +1,35 @@
+Removed Option: `typo3UploadFieldsInTopOfEB`
+--------------------------------------------
+
+In the latest version of TYPO3, the `typo3UploadFieldsInTopOfEB` option has been removed. This change was necessitated
```
`latest version`? It was removed somewhere before TYPO3 6.2.

sfroemkenjw commented on Resources/Private/Language/ExtConf.xlf:
```diff
@@ -60,12 +60,6 @@
 					var/log/.
 				</source>
 			</trans-unit>
-			<trans-unit id="enableLiveSearchPerformanceForAdmins">
```
Missing removal of typo3UploadFieldsInTopOfEB

sfroemkenjw commented on Resources/Private/Language/de.ExtConf.xlf:
```diff
@@ -47,11 +47,6 @@
 				<source>Enable Caching Framework Logger: Hooks into the TYPO3 caching framework and parses all data using expression records to be created on the root page (PID: 0). If these match, a log entry is created in var/log/.</source>
 				<target>Caching Framework Logger aktivieren: Klinkt sich in das TYPO3 Caching Framework ein und analysiert alle Daten anhand von Expression Records, die auf der Root-Seite (PID: 0) zu erstellen sind. Wenn diese übereinstimmen, wird ein Protokolleintrag in var/log/ erzeugt</target>
 			</trans-unit>
-			<trans-unit id="enableLiveSearchPerformanceForAdmins">
```
Missing removal of typo3UploadFieldsInTopOfEB

sfroemkenjw commented on Resources/Private/Templates/Solr/List.html:
```diff
@@ -1,80 +1,90 @@
-			xmlns:c="http://typo3.org/ns/TYPO3\CMS\Core\ViewHelpers"
-			data-namespace-typo3-fluid="true">
+      xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers"
+      xmlns:t="http://typo3.org/ns/JWeiland\Jwtools2\ViewHelpers"
```
Slashes, not backslashes

sfroemkenjw commented on Resources/Private/Templates/Solr/List.html:
```diff
@@ -1,80 +1,90 @@
-			data-namespace-typo3-fluid="true">
+      xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers"
+      xmlns:t="http://typo3.org/ns/JWeiland\Jwtools2\ViewHelpers"
+      xmlns:c="http://typo3.org/ns/TYPO3\CMS\Core\ViewHelpers"
```
Slashes, not backslashes

sfroemkenjw commented on Resources/Private/Templates/Solr/Show.html:
```diff
@@ -1,34 +1,38 @@
-			xmlns:t="http://typo3.org/ns/JWeiland\Jwtools2\ViewHelpers"
-			data-namespace-typo3-fluid="true">
+      xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers"
+      xmlns:t="http://typo3.org/ns/JWeiland\Jwtools2\ViewHelpers"
```
Slashes, not backslashes

sfroemkenjw commented on Resources/Private/Templates/Solr/ShowClearFullIndexForm.html:
```diff
@@ -1,83 +1,87 @@
-			xmlns:c="http://typo3.org/ns/TYPO3\CMS\Core\ViewHelpers"
-			data-namespace-typo3-fluid="true">
+      xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers"
+      xmlns:c="http://typo3.org/ns/TYPO3\CMS\Core\ViewHelpers"
```
Slashes, not backslashes

sfroemkenjw commented on Resources/Private/Templates/Tools/Overview.html:
```diff
@@ -1,15 +1,14 @@
-<html lang="en"
-			xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers"
-			data-namespace-typo3-fluid="true">
+<html data-namespace-typo3-fluid="true" xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers">
```
Use new lines incl. lang attribute

sfroemkenjw commented on Tests/Functional/Hooks/InitializeStdWrapTest.php:
```diff
@@ -11,54 +11,32 @@
         parent::setUp();
 
-        $this->extensionConfigurationProphecy = $this->prophesize(ExtensionConfiguration::class);
+        $this->extensionConfigurationMock = $this->getAccessibleMock(ExtensionConfiguration::class);
```
A normal `getMock` or `createMock` should be enough as you don't make use of all the specific _set methods

sfroemkenjw commented on ext_conf_template.txt:
```diff
@@ -2,8 +2,12 @@
 # cat=typo3; type=boolean; label=LLL:EXT:jwtools2/Resources/Private/Language/ExtConf.xlf:typo3TransferTypoScriptCurrent
 typo3TransferTypoScriptCurrent = 0
+
+# @todo: rafactor this for version 12
```
We are in version 12

sfroemkenjw commented on ext_conf_template.txt:
```diff
@@ -2,8 +2,12 @@
 # cat=typo3; type=boolean; label=LLL:EXT:jwtools2/Resources/Private/Language/ExtConf.xlf:typo3UploadFieldsInTopOfEB
 typo3UploadFieldsInTopOfEB = 0
+
+# @todo: rafactor this for version 12
```
We are in version 12

hojalatheef committed:
[TASK] CR Fixes

- Removed typo3UploadFieldsInTopOfEB configuration related references
- Services.yaml configuration for DI of SchedulerTaskRepository
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549583142
- Rename method to getQueryResultForUsers
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549588058
- https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549590280
- Removed space
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549626275
- Removed #AsController
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549639251
- Changed function name
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549655337
- Removed count
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549602310
- Moved $yesterday inside the function and removed from function arguments
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549621061
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549621556
- Changed comment style
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549623525
- Removed float return type
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549640894
- Changed variable type
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549657556
- Replaced substr with str_ends_with
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549813428
- Remove unwanted casting as the variable is already casted
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549816431
- Deleted ViewHelper
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549823686
- Missing space inserted
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549826892
- Documentation Corrections
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549832518
- Changed backslashes to forward slashes
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549845172
- Changed backslashes to forward slashes
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549843633
- Add lang attribute
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549848126
- Removed todo comments as it was already done
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549863404
- Changed getAccessibleMock to createMock
  https://github.com/jweiland-net/jwtools2/pull/74#discussion_r1549857896

hojalatheef committed:
Loaded scheduler extension as it is not loaded by default

hojalatheef committed:
[TASK] Removed nullable in property declaration

sfroemkenjw commented on Classes/Backend/SolrDocHeader.php:
```diff
@@ -32,12 +32,6 @@ public function __construct(
-        if (!($this->uriBuilder instanceof UriBuilder)) {
-            $this->uriBuilder = GeneralUtility::makeInstance(UriBuilder::class);
-        }
-        $this->uriBuilder->setRequest($this->request);
```
I understand the 3 lines above, but why you have removed this line?

hojalatheef replied:
@sfroemkenjw Because the request is already set with dependency injection.

sfroemkenjw replied:
Can you please show me the line where the Request class was included into UriBuilder via DependencyInjection?

sfroemkenjw commented on Classes/Command/StatusReportCommand.php:
```diff
@@ -40,9 +40,7 @@ class StatusReportCommand extends Command
-        parent::__construct();
-
-        $this->taskRepository = GeneralUtility::makeInstance(SchedulerTaskRepository::class);
+        parent::__construct('Start Report Command');
```
This was not my idea. And why you have removed the taskRepository initialization?
Please change to:

```
public function __construct(?string $name = null)
{
    parent::__construct($name);

    $this->taskRepository = GeneralUtility::makeInstance(SchedulerTaskRepository::class);
}

sfroemkenjw replied:
I just see you have implemented taskRepo with a setter. So, in that case the constructor can be removed completely

hojalatheef replied:
okay

sfroemkenjw commented on Classes/Database/Query/QueryGenerator.php:
```diff
@@ -24,7 +24,7 @@ class QueryGenerator
      * @throws Exception
      */
-    public function getTreeList($id, $depth, $begin = 0, $permClause = ''): float|int|string
+    public function getTreeList($id, $depth, $begin = 0, $permClause = ''): int|string
```
It returns INT, if there is just one numer available and it returns string, if there are multiple numbers available. Sure, this method is a copy of TYPO3 source and they have to be careful changing return types. But this one here is our own method, we can adopt it to our needs as we want. I prefer to set string as return type in general.

Another solution is:
return type array and you do a GeneralUtility::intExplode() here. In that case all floating values were migrated, too.

sfroemkenjw commented on Resources/Private/Templates/Tools/Overview.html:
```diff
@@ -1,4 +1,6 @@
-<html data-namespace-typo3-fluid="true" xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers">
+<html lang="en"
+      data-namespace-typo3-fluid="true"
```
Always move `data-namespace-typo3-fluid` to last position. Why?
In most cases you have to add one or two new ViewHelper declarations. It's easier to mark the f namespace line and press CONTROL+D (duble the line).
In your solution I additionally have to move cursor to end of line and remove a duplicated `>`

sfroemkenjw commented on Tests/Functional/Hooks/InitializeStdWrapTest.php:
```diff
@@ -25,6 +25,10 @@ class InitializeStdWrapTest extends FunctionalTestCase
     protected ExtensionConfiguration | MockObject $extensionConfigurationMock;
 
+    protected array $coreExtensionsToLoad = [
+        'typo3/cms-scheduler'
```
Missing `,` at end of file. Why does php-cs-fixer does not tell you about? Is `Tests` directory in path declaration?

sfroemkenjw commented on q:
```diff
@@ -0,0 +1,15 @@
+[TASK] CR Fixes
```
I don't know that file. Please explain (and add it to .gitattributes or .gitignore) or remove

hojalatheef committed:
Removed constructor as the DI used with a setter method

sfroemkenjw commented on Classes/Controller/SolrController.php:
```diff
@@ -16,112 +16,82 @@
+use JWeiland\Jwtools2\Traits\InjectSchedulerRepositoryTrait;
+use JWeiland\Jwtools2\Traits\InjectSolrRepositoryTrait;
+use Psr\Http\Message\ResponseInterface;
+use TYPO3\CMS\Backend\Attribute\AsController;
```
AsController was not needed anymore. Why does php-cs-fixer does not found that?

hojalatheef committed:
Change return type to string for getTreeList function

hojalatheef committed:
[TASK] Template cleanup

- Cleanup templates which are not required anymore
- Formating Fluid templates with space instead of tab intendation

Resolves: #
Releases: main, 12.4

hojalatheef committed:
Missing comma issue

hojalatheef committed:
Removed unwanted file created as part of commit error

hojalatheef committed:
Removed annotations which are not used

hojalatheef committed:
Removed addHelpButton implementation as it is no more valid and render empty.
